src/test/rpc/AccountCurrencies_test.cpp

```diff
@@ -0,0 +1,175 @@
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <ripple/beast/unit_test.h>
+
+namespace ripple {
+
+class AccountCurrencies_test : public beast::unit_test::suite
+{
+    void
+    testBadInput ()
+    {
+        testcase ("Bad input to account_currencies");
+
+        using namespace test::jtx;
+        Env env {*this};
+
+        auto const alice = Account {"alice"};
+        env.fund (XRP(10000), alice);
+        env.close ();
+
+        { // invalid ledger (hash)
+            Json::Value params;
+            params[jss::ledger_hash] = 1;
+            auto const result = env.rpc ("json", "account_currencies",
+                boost::lexical_cast<std::string>(params)) [jss::result];
+            BEAST_EXPECT (result[jss::error] == "invalidParams");
+            BEAST_EXPECT (result[jss::error_message] ==
+                "ledgerHashNotString");
+        }
+
+        { // missing account field
+            auto const result =
+                env.rpc ("json", "account_currencies", "{}") [jss::result];
+            BEAST_EXPECT (result[jss::error] == "invalidParams");
+            BEAST_EXPECT (result[jss::error_message] ==
+                "Missing field 'account'.");
+        }
+
+        { // strict mode, invalid bitcoin token
+            Json::Value params;
+            params[jss::account] = "llIIOO"; //these are invalid in bitcoin alphabet
+            params[jss::strict] = true;
+            auto const result = env.rpc ("json", "account_currencies",
+                boost::lexical_cast<std::string>(params)) [jss::result];
+            BEAST_EXPECT (result[jss::error] == "actMalformed");
+            BEAST_EXPECT (result[jss::error_message] ==
+                "Account malformed.");
+        }
+
+        { // strict mode, using properly formatted bitcoin token
+            Json::Value params;
+            params[jss::account] = base58EncodeTokenBitcoin (
+                TOKEN_ACCOUNT_ID, alice.id().data(), alice.id().size());
+            params[jss::strict] = true;
+            auto const result = env.rpc ("json", "account_currencies",
+                boost::lexical_cast<std::string>(params)) [jss::result];
+            BEAST_EXPECT (result[jss::error] == "actBitcoin");
+            BEAST_EXPECT (result[jss::error_message] ==
+                "Account is bitcoin address.");
+        }
+
```

Comment:
Good point - so that means we either have a doc bug or an implementation bug. I can't see any reason that we shouldn't just return an error for a nonexistent account in this API, but paging @nikb and @joelkatz for their take.